src/System.Slices/System/Span.cs

```diff
@@ -400,6 +400,36 @@ public override bool Equals(object obj)
             return false;
         }
 
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsSliceOf(Span<T> parentSpan)
+        {
+            var elementSize = PtrUtils.SizeOf<T>();
+            return Object == parentSpan.Object // either both pointers (both null), or same array instance
+                && parentSpan.Offset.ToUInt64() <= Offset.ToUInt64() // parent must start earlier (or equal start)
```

Comment:
fix pushed, thanks; can only really apply to the left hand bound check, due to the math issue